File: api/export.js
const ExcelJS = require('exceljs');
const { db } = require('./_db');
const path = require('path');
const fs = require('fs');

module.exports = async (req, res) => {
    try {
        const teamId = req.query.id;
        if (!teamId) {
            return res.status(400).json({ error: "Missing team id" });
        }

        const { rows: teamRows } = await db.query(
            "SELECT id, name, filename, sheet_name FROM teams WHERE id = $1",
            [teamId]
        );

        if (teamRows.length === 0) {
            return res.status(404).json({ error: "Team not found" });
        }

        const team = teamRows[0];
        const { rows: players } = await db.query(
            "SELECT excel_row, photo_data FROM players WHERE team_id = $1 AND photo_data IS NOT NULL",
            [teamId]
        );

        const templatePath = path.join(process.cwd(), 'LISTAS DE BUENA FE AFA SOMOS TODOS 2026', team.filename);
        
        if (!fs.existsSync(templatePath)) {
            return res.status(500).json({ error: `Template file not found: ${team.filename}` });
        }

        const workbook = new ExcelJS.Workbook();
        await workbook.xlsx.readFile(templatePath);
        
        const ws = workbook.getWorksheet(team.sheet_name) || workbook.worksheets[0];

        // Ensure column G is wide enough
        ws.getColumn(7).width = 25;

        for (const player of players) {
            if (!player.photo_data) continue;

            const rowIdx = parseInt(player.excel_row);
            const row = ws.getRow(rowIdx);
            row.height = 95;

            const imageId = workbook.addImage({
                buffer: player.photo_data,
                extension: 'jpeg',
            });

            // G = 7th column. Anchor is 0-indexed, so G is col 6.
            ws.addImage(imageId, {
                tl: { col: 6.1, row: rowIdx - 0.9 },
                ext: { width: 130, height: 170 }
            });
        }

        res.setHeader(
            'Content-Type',
            'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet'
        );
        res.setHeader(
            'Content-Disposition',
            `attachment; filename="${team.name}_FOTOS.xlsx"`
        );

        const buffer = await workbook.xlsx.writeBuffer();
        res.send(buffer);

    } catch (error) {
        console.error("Export error:", error);
        res.status(500).json({ error: error.message });
    }
};

File: api/export.py
from http.server import BaseHTTPRequestHandler
from urllib.parse import urlparse, parse_qs
import psycopg2
import os
import openpyxl
from openpyxl.drawing.image import Image as OpenpyxlImage
from PIL import Image as PILImage
import io

class handler(BaseHTTPRequestHandler):
    def do_GET(self):
        try:
            parsed_path = urlparse(self.path)
            query = parse_qs(parsed_path.query)
            team_id = query.get('id', [None])[0]
            
            if not team_id:
                self.send_error(400, "Missing team id")
                return
                
            team_id = int(team_id)
            
            # Connect to Neon Postgres
            conn = psycopg2.connect(os.environ.get('DATABASE_URL'))
            cur = conn.cursor()
            
            # Get team info
            cur.execute("SELECT id, name, filename, sheet_name FROM teams WHERE id = %s", (team_id,))
            team = cur.fetchone()
            
            if not team:
                self.send_error(404, "Team not found")
                cur.close()
                conn.close()
                return
                
            t_name, t_filename, t_sheet = team[1], team[2], team[3]
            
            # Get players with photos
            cur.execute("SELECT excel_row, photo_data FROM players WHERE team_id = %s AND photo_data IS NOT NULL", (team_id,))
            players = cur.fetchall()
            
            cur.close()
            conn.close()
            
            # Find template
            template_path = os.path.join(os.getcwd(), 'LISTAS DE BUENA FE AFA SOMOS TODOS 2026', t_filename)
            if not os.path.exists(template_path):
                self.send_error(500, f"Template file not found: {t_filename}")
                return
                
            # Process Excel
            wb = openpyxl.load_workbook(template_path)
            ws = wb[t_sheet] if t_sheet in wb.sheetnames else wb.active
            
            # Clear old images (clean up the template)
            ws._images = []
            ws.column_dimensions['G'].width = 25
            
            for p_row, photo_bytes in players:
                if not photo_bytes:
                    continue
                
                # In PostgreSQL, bytea might be returned as memoryview or bytes
                if isinstance(photo_bytes, memoryview):
                    photo_bytes = photo_bytes.tobytes()
                    
                # Set row height
                ws.row_dimensions[p_row].height = 95
                
                # Resize image
                img_stream = io.BytesIO(photo_bytes)
                pil_img = PILImage.open(img_stream)
                
                target_w, target_h = 130, 170
                orig_w, orig_h = pil_img.size
                ratio = min(target_w / orig_w, target_h / orig_h)
                new_w, new_h = int(orig_w * ratio), int(orig_h * ratio)
                
                pil_img = pil_img.resize((new_w, new_h), PILImage.Resampling.LANCZOS)
                if pil_img.mode in ("RGBA", "P"):
                    pil_img = pil_img.convert("RGB")
                    
                resized_stream = io.BytesIO()
                pil_img.save(resized_stream, format="JPEG", quality=90)
                resized_stream.seek(0)
                
                xl_img = OpenpyxlImage(resized_stream)
                
                # Match original oneCellAnchor behavior
                col_idx = 6 # G (0-indexed A=0)
                row_idx = p_row - 1 # 0-indexed
                
                marker = openpyxl.drawing.spreadsheet_drawing.OneCellAnchor(
                    _from=openpyxl.drawing.spreadsheet_drawing.AnchorMarker(col=col_idx, colOff=0, row=row_idx, rowOff=0),
                    ext=openpyxl.drawing.xdr.XDRPositiveSize2D(cx=new_w * 9525, cy=new_h * 9525)
                )
                xl_img.anchor = marker
                ws.add_image(xl_img)
                
            # Save to buffer
            out_stream = io.BytesIO()
            wb.save(out_stream)
            out_stream.seek(0)
            
            # Send response
            self.send_response(200)
            self.send_header('Content-type', 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet')
            self.send_header('Content-Disposition', f'attachment; filename="{t_name}_FOTOS.xlsx"')
            self.end_headers()
            
            self.wfile.write(out_stream.read())
            
        except Exception as e:
            self.send_response(500)
            self.send_header('Content-type', 'application/json')
            self.end_headers()
            import json
            self.wfile.write(json.dumps({"error": str(e)}).encode())
            
